web/packages/teleport/src/components/TdpClientCanvas/TdpClientCanvas.tsx

```diff
@@ -110,20 +108,29 @@ function TdpClientCanvas(props: Props) {
         hotspot_y?: number;
       }) => {
         if (typeof pointer.data === 'boolean') {
-          if (pointer.data) {
-            canvas.style.cursor = previousCursor.current;
-          } else {
-            previousCursor.current = canvas.style.cursor;
-            canvas.style.cursor = 'none';
-          }
+          canvas.style.cursor = pointer.data ? 'default' : 'none';
           return;
         }
-        const cursor = document.createElement('canvas');
+        let cursor = document.createElement('canvas');
         cursor.width = pointer.data.width;
         cursor.height = pointer.data.height;
         cursor
           .getContext('2d', { colorSpace: pointer.data.colorSpace })
           .putImageData(pointer.data, 0, 0);
+        if (pointer.data.width > 32 || pointer.data.height > 32) {
+          // scale the cursor down to at most 32px - max size fully supported by browsers
```

Comment:
We currently tell the server we can handle up to 384x384px cursors. Maybe it would be better not to advertise for that?

https://github.com/Devolutions/IronRDP/blob/1f067069bab87e15b758e6e5ecc9a0330bc49639/crates/ironrdp-connector/src/connection_activation.rs#L336-L341